src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -26,33 +27,50 @@ object Scope {
     def getChildrenList(ast: Ast): Seq[Ast] = Option(ast.getAttribute("body")) collect {
       case list: AstList => list.asScala.toVector
     } getOrElse Seq.empty collect {
-      case ast: Ast => ast
+      case node: Ast
+        if node.getName == AstNodeName.Call ||
+          node.getName == AstNodeName.Scatter =>
+        node
+    }
```

Comment:
Using the magic of [`collect`](http://danielwestheide.com/blog/2012/12/12/the-neophytes-guide-to-scala-part-4-pattern-matching-anonymous-functions.html) during the scope tree creation, this partial function also filters out nodes like `AstNodeName.Declarations`, and only includes child nodes of `Call` and `Scatter`. Even `Workflow` is removed here.

Later in `generateScopeTree`, the filtering reduced our expected values, so we can just throw the idiomatic `scala.MatchError` if we ever `match` an unexpected `node`. We then don't need a `case _ => None` to cleanup after us, nor need to return `Option[Scope]` anymore, and can just recursively return `Scope`. Also, because we moved `Workflow` outside the recursive iteration, `generateScopeTree` no longer needs to match on `AstNodeName.Workflow` either.

In a previous version, I also went down a path with type checked `ParentScope` and `ChildScope`, where perhaps even a `Call` would cause a blow up if someone tried to set its children. But, I needed to cauterize this particular PR and checkpoint, and will save that possibly for another day, unless unexpectedly reviewers do want that now.